### PR TITLE
完善基于ebpf观测用户态函数时延项目：利用ptrace机制解决短时间程序的观测

### PR DESCRIPTION
**Pull request by jyf111**
# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

1. 利用ptrace机制解决短时间程序的观测
2. 完善了输出形式
3. 添加命令行参数解析
4. 支持输入需要观测的已运行进程的PID
5. 进行了更多测试

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: eBPF_Supermarket/User_Function_Tracer/README.md [hunk 1]
```diff
@@ -25,9 +36,5 @@ build/utrace PROGRAM|PID
 + 不同于`perf`, `gprof`等性能分析工具，`eBPF-utrace`输出准确的函数调用时延，而不是基于perf_event()的采样方式。
 
 ### TODO
-- [] short live process
-- [] arg parser
-- [] input pid
-- [] more tests
-- [] colorful log
-- [] multithread
+- simplify c++ symbols
```
Comment (helight) [inline eBPF_Supermarket/User_Function_Tracer/README.md]: 已近具备的能力也可以罗列一下

File: eBPF_Supermarket/User_Function_Tracer/src/symbol.c [hunk 2]
```diff
@@ -18,15 +18,16 @@
 
 #include "symbol.h"
```
Comment (helight) [inline eBPF_Supermarket/User_Function_Tracer/src/symbol.c:20]: 文件头没有加上

Comment (jyf111) [reply eBPF_Supermarket/User_Function_Tracer/src/symbol.c:20]: 是有文件头的

File: eBPF_Supermarket/User_Function_Tracer/test/strcpy.c [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#include <stdlib.h>
```
Comment (helight) [inline eBPF_Supermarket/User_Function_Tracer/test/strcpy.c:19]: 这些文件的头注释也加上一下，只要是我们开发的文件都加上

File: eBPF_Supermarket/User_Function_Tracer/test/mmap.c [hunk 4]
```diff
@@ -1,4 +1,5 @@
 #include <fcntl.h>
+#include <stdio.h>
```
Comment (helight) [inline eBPF_Supermarket/User_Function_Tracer/test/mmap.c:20]: 上面这个 mmap 二进制文件不要提交

**Review by helight — APPROVED** [review]
LGTM

**Review by chenamy2017 — APPROVED** [review]

**Pull request merged**